From: Robie Basak <robie.basak@oss.qualcomm.com>
Description: use versioned .so.1 names when calling dlopen()
 Since Debian does not ship the .so symlink in its runtime library packages,
 they cannot be used at runtime. Explicitly referencing the required ABI version
 is the correct way to do this, to avoid problems with any future ABI
 incompatilities.
Bug: https://github.com/qualcomm/fastrpc/issues/230
Last-Update: 2025-09-01

--- a/src/adsprpcd.c
+++ b/src/adsprpcd.c
@@ -15,10 +15,10 @@
 
 
 #ifndef ADSP_DEFAULT_LISTENER_NAME
-#define ADSP_DEFAULT_LISTENER_NAME "libadsp_default_listener.so"
+#define ADSP_DEFAULT_LISTENER_NAME "libadsp_default_listener.so.1"
 #endif
 #ifndef ADSP_LIBHIDL_NAME
-#define ADSP_LIBHIDL_NAME "libhidlbase.so"
+#define ADSP_LIBHIDL_NAME "libhidlbase.so.1"
 #endif
 
 typedef int (*adsp_default_listener_start_t)(int argc, char *argv[]);
--- a/src/cdsprpcd.c
+++ b/src/cdsprpcd.c
@@ -14,10 +14,10 @@
 #include <unistd.h>
 
 #ifndef CDSP_DEFAULT_LISTENER_NAME
-#define CDSP_DEFAULT_LISTENER_NAME "libcdsp_default_listener.so"
+#define CDSP_DEFAULT_LISTENER_NAME "libcdsp_default_listener.so.1"
 #endif
 #ifndef CDSP_LIBHIDL_NAME
-#define CDSP_LIBHIDL_NAME "libhidlbase.so"
+#define CDSP_LIBHIDL_NAME "libhidlbase.so.1"
 #endif
 
 typedef int (*adsp_default_listener_start_t)(int argc, char *argv[]);
--- a/src/sdsprpcd.c
+++ b/src/sdsprpcd.c
@@ -13,7 +13,7 @@
 
 
 #ifndef SDSP_DEFAULT_LISTENER_NAME
-#define SDSP_DEFAULT_LISTENER_NAME "libsdsp_default_listener.so"
+#define SDSP_DEFAULT_LISTENER_NAME "libsdsp_default_listener.so.1"
 #endif
 
 typedef int (*adsp_default_listener_start_t)(int argc, char *argv[]);
